backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/UserProfileController.cs

```diff
@@ -0,0 +1,32 @@
+﻿using EasySpeak.Core.BLL.Interfaces;
+using EasySpeak.Core.Common.DTO.UploadFile;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EasySpeak.Core.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UserProfileController : ControllerBase
+    {
+        private readonly IEasySpeakFileService _easySpeakFileService;
+        private readonly IFirebaseAuthService _firebaseAuthService;
+
+        public UserProfileController(IEasySpeakFileService easySpeakFileService, IFirebaseAuthService firebaseAuthService)
+        {
+            _easySpeakFileService = easySpeakFileService;
+            _firebaseAuthService = firebaseAuthService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> UploadAvatar(IFormFile file)
+        {
+            var fileDto = new NewEasySpeakFileDto()
+            {
+                Stream = file.OpenReadStream(),
+                FileName = file.FileName
+            };
+            var res = await _easySpeakFileService.AddFileAsync(fileDto);
```

Comment:
Can you move this call to like UserService, where at the end this file will be assigned to user as a profile picture
Maybe you will need to create migration

Then call userService.UpdateProfilePhoto from this controller